esim-cloud-backend/Dockerfile

```diff
@@ -34,7 +34,41 @@ RUN curl -fSL https://github.com/imr/ngspice/archive/ngspice-$NGSPICE_VERSION.ta
 RUN apk add --no-cache libxt jpeg-dev zlib-dev libxaw-dev libx11-dev libtool mariadb-connector-c-dev libffi-dev postgresql-dev
 
 COPY requirements.txt /code/
-RUN apk add --no-cache --virtual .build-deps gcc python3-dev musl-dev mariadb-dev musl-dev\
+RUN apk add --no-cache mariadb-connector-c-dev ;\
+    apk add --update alpine-sdk && \
+    apk add libffi-dev openssl-dev && \
+    apk --no-cache --update add build-base ;\
+    apk add --no-cache --virtual .build-deps gcc python3-dev musl-dev mariadb-dev\
     && pip install --upgrade pip\
     && pip install --trusted-host pypi.python.org -r requirements.txt \
     && apk del .build-deps
+
+
+RUN apk add \
```

Comment:
Can we combine this with the requirements installed on line 34 ?
Also let's keep these static dependencies above the pip install  for better image caching

Reply:
I did that on purpose, just to separate the requirements for Arduino backend,

Reply:
docker re-builds all the layers below the line where a change is detected , so if requirements.txt is modified , all the arduino dependencies are built and installed again , instead of just the python requirements being updated while building. 
This increases the build time and size significantly.

Reply:
ok I will move the stuff above.